subprojects/jacoco/src/integTest/groovy/org/gradle/testing/jacoco/plugins/rules/JacocoPluginCoverageVerificationIntegrationTest.groovy

```diff
@@ -0,0 +1,352 @@
+        succeeds TEST_AND_JACOCO_COVERAGE_VERIFICATION_TASK_PATHS
+
+        then:
+        executedAndNotSkipped(TEST_AND_JACOCO_COVERAGE_VERIFICATION_TASK_PATHS)
+    }
+
+    def "can ignore failures"() {
+        given:
+        buildFile << """
+            jacocoTestCoverageVerification {
+                violationRules {
+                    failOnViolation = true
+
+                    rule {
+                        $Insufficient.LINE_METRIC_COVERED_RATIO
+                    }
+                }
+            }
+        """
+
+        when:
+        succeeds TEST_AND_JACOCO_COVERAGE_VERIFICATION_TASK_PATHS
+
+        then:
+        executedAndNotSkipped(TEST_AND_JACOCO_COVERAGE_VERIFICATION_TASK_PATHS)
+        errorOutput.contains("Rule violated for bundle $testDirectory.name: lines covered ratio is 1.0, but expected maximum is 0.5")
+    }
+
+    @Unroll
+    def "can define same rules for multiple report tasks #tasksPaths"() {
+        given:
+        javaProjectUnderTest.writeIntegrationTestSourceFiles()
+
+        buildFile << """
+            tasks.withType(JacocoCoverageVerification) {
+                violationRules {
+                    rule {
+                        $Insufficient.LINE_METRIC_COVERED_RATIO
+                    }
+                }
+            }
+        """
+
+        when:
+        fails tasksPaths
+
+        then:
+        executedAndNotSkipped(tasksPaths)
+        errorOutput.contains("Rule violated for bundle $testDirectory.name: lines covered ratio is 1.0, but expected maximum is 0.5")
+
+        where:
+        tasksPaths << [TEST_AND_JACOCO_COVERAGE_VERIFICATION_TASK_PATHS, INTEG_TEST_AND_JACOCO_COVERAGE_VERIFICATION_TASK_PATHS]
+    }
+
+    @Unroll
+    def "can define different rules for multiple report tasks #tasksPaths"() {
+        given:
+        javaProjectUnderTest.writeIntegrationTestSourceFiles()
+
+        buildFile << """
+            $reportTaskName {
+                violationRules {
+                    rule {
+                        $limit
+                    }
+                }
+            }
+        """
+
+        when:
+        fails tasksPaths
+
+        then:
+        executedAndNotSkipped(tasksPaths)
+        errorOutput.contains("Rule violated for bundle $testDirectory.name: $errorMessage")
+
+        where:
+        tasksPaths                                             | reportTaskName                              | limit                                  | errorMessage
+        TEST_AND_JACOCO_COVERAGE_VERIFICATION_TASK_PATHS       | 'jacocoTestCoverageVerification'            | Insufficient.LINE_METRIC_COVERED_RATIO | 'lines covered ratio is 1.0, but expected maximum is 0.5'
+        INTEG_TEST_AND_JACOCO_COVERAGE_VERIFICATION_TASK_PATHS | 'jacocoIntegrationTestCoverageVerification' | Insufficient.CLASS_METRIC_MISSED_COUNT | 'classes missed count is 0.0, but expected minimum is 0.5'
+    }
+
+    def "task is never UP-TO-DATE as it does not define any outputs"() {
```

Comment:
I get your point but I'd like to avoid widening the scope of this feature. At the moment I cannot imagine what the content of the text file would be. We could write the configured rules to the file but there's no information we can get about the execution of validation we could capture. IMHO if a user wants to have that behavior then it can easily be added after the fact by defining an output.